testsuite/integration/web/src/test/java/org/jboss/as/test/integration/web/handlers/ForwardedHandlerTestCase.java

```diff
@@ -0,0 +1,124 @@
+package org.jboss.as.test.integration.web.handlers;
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SUBSYSTEM;
+import static org.junit.Assert.assertEquals;
+
+import java.io.IOException;
+import java.net.URL;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.StatusLine;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.OperateOnDeployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.as.arquillian.api.ContainerResource;
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.operations.common.Util;
+import org.jboss.as.test.integration.management.ManagementOperations;
+import org.jboss.dmr.ModelNode;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Tests the use of undertow-handlers.conf
+ *
+ * @author Stuart Douglas
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class ForwardedHandlerTestCase {
+
+    private static final String FORWARDED_HANDLER = "forwarded-handler";
+    private static final String FORWARDED_HANDLER_NO_UT_HANDLERS = "forwarded-handler-no-ut-handlers";
+
+    private static final String FORWARDER_HANDLER_NAME = "forwarded";
+
+    private static final PathAddress FORWARDER_CONF_ADDR = PathAddress.pathAddress().append(SUBSYSTEM, "undertow")
+            .append("configuration", "filter").append("expression-filter", "ff");
+    private static final PathAddress FORWARDER_FILTER_REF_ADDR = PathAddress.pathAddress().append(SUBSYSTEM, "undertow")
+            .append("server", "default-server").append("host", "default-host").append("filter-ref", "ff");
+
+    private static final String JBOSS_WEB_TEXT = "<?xml version=\"1.0\"?>\n" +
+            "<jboss-web>\n" +
+            "    <http-handler>\n" +
+            "        <class-name>org.jboss.as.test.integration.web.handlers.ForwardedTestHelperHandler</class-name>\n" +
```

Comment:
This should really be testing with a Servlet, rather than an Undertow handler as that is what 99%+ of people will actually be using.

Reply:
Thanks - I'll extended this so also servlet is utilized.